lib/utils/prometheus.go

```diff
@@ -0,0 +1,21 @@
+package utils
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// RegisterPrometheusCollectors is a wrapper around prometheus.Register that
+// - ignores equal or re-registered collectors
+// - return an error if a collector does not fulfill the consistency and
+//   uniqueness criteria
+func RegisterPrometheusCollectors(collectors ...prometheus.Collector) error {
```

Comment:
If I understand this correctly, the teleport team would need to be careful with things like re-init a service that registry prometheus metrics, since they might get dropped due to a previous registration of the same metric? And if it were to happen the metrics would just be missing, since they're being recorded in an unregistered object?

Reply:
No, re-registering is fine. but changing a metric type is not.
So if someone is re-registering a metric with the same name under a different type, we'd get an error here, while teleport would panic in the previous implementation.